### PR TITLE
Move the entire current state to Zustand store

### LINKED ISSUE #6757: [refactor]: Move currentState from Editor component to Zustand store
### What area of the codebase is to be refactored?

Editor.tsx, appUtils.js

### What are the potential advantages of this proposed refactoring?

This refactor will help in reducing prop drilling at multiple nested level components and also to reduce multiple re-renders.

### PR DESCRIPTION
**Pull request by nakulnagargade**
Resolves #6757

### CODE REVIEW

File: frontend/src/Editor/LeftSidebar/SidebarInspector.jsx [hunk 1]
```diff
@@ -16,7 +17,8 @@ const staticDataSources = [
 
 export const LeftSidebarInspector = ({
   darkMode,
-  currentState,
+  selectedSidebarItem,
+  setSelectedSidebarItem,
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/LeftSidebar/SidebarInspector.jsx:21]: If it is not used anywhere we can remove it

File: frontend/src/Editor/LeftSidebar/index.jsx [hunk 2]
```diff
@@ -225,7 +223,7 @@ export const LeftSidebar = forwardRef((props, ref) => {
       <LeftSidebarInspector
         darkMode={darkMode}
         selectedSidebarItem={selectedSidebarItem}
-        currentState={currentState}
+        setSelectedSidebarItem={handleSelectedSidebarItem}
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/LeftSidebar/index.jsx:226]: Don't think this is being used in the component. Please check and remove it if not needed

File: frontend/src/Editor/Editor.jsx [hunk 3]
```diff
@@ -112,22 +113,6 @@ class EditorComponent extends React.Component {
       currentLayout: 'desktop',
       deviceWindowWidth: 450,
       appDefinition: this.defaultDefinition,
-      currentState: {
-        queries: {},
-        components: {},
-        globals: {
-          theme: { name: props.darkMode ? 'dark' : 'light' },
-          urlparams: JSON.parse(JSON.stringify(queryString.parse(props.location.search))),
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Editor.jsx:120]: Seems like we're not updating the store with these values. It should be updated during the mount

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@nakulnagargade Requested few code suggestions and I'm not able to test the app. When I try to load or create a new app, the page crashes. Please take a look

File: frontend/src/_helpers/appUtils.js [hunk 4]
```diff
@@ -1643,24 +1566,21 @@ export const computeQueryState = (queries, _ref) => {
...
     } else {
       queryState[query.name] = {
         ...DataSourceTypes.find((source) => source.kind === query.kind).exposedVariables,
         kind: DataSourceTypes.find((source) => source.kind === query.kind).kind,
-        ..._ref.state.currentState.queries[query.name],
+        ...useCurrentStateStore.getState().queries[query.name],
       };
     }
   });
   const hasDiffQueryState = !_.isEqual(_ref.state?.currentState?.queries, queryState);
```
[the remark this answers is not in the document]
Comment (nakulnagargade) [reply frontend/src/_helpers/appUtils.js:1579]: Done

File: frontend/src/Editor/LeftSidebar/SidebarInspector.jsx (hunk 1 above)
Comment (nakulnagargade) [reply frontend/src/Editor/LeftSidebar/SidebarInspector.jsx]: Done

File: frontend/src/Editor/LeftSidebar/index.jsx (hunk 2 above)
Comment (nakulnagargade) [reply frontend/src/Editor/LeftSidebar/index.jsx]: Not used, removed.

File: frontend/src/Editor/Editor.jsx (hunk 3 above)
Comment (nakulnagargade) [reply frontend/src/Editor/Editor.jsx:120]: Yes, Done

**Review by kavinvenkatachalam — APPROVED** [review]

File: frontend/src/Editor/Inspector/ActionConfigurationPanels/SwitchPage.jsx [hunk 5]
```diff
@@ -3,13 +3,15 @@ import Select from '@/_ui/Select';
...
+import { useCurrentState } from '@/_stores/currentStateStore';
 
-export function SwitchPage({ getPages, currentState, event, handlerChanged, eventIndex, darkMode }) {
+export function SwitchPage({ getPages, event, handlerChanged, eventIndex, darkMode }) {
   const queryParamChangeHandler = (index, key, value) => {
     event.queryParams[index][key] = value;
     handlerChanged(eventIndex, 'queryParams', event.queryParams);
   };
   const { t } = useTranslation();
+  const currentState = useCurrentState();
```
Comment (johnsoncherian) [inline frontend/src/Editor/Inspector/ActionConfigurationPanels/SwitchPage.jsx:14]: seems like we are using currentState in this component only to pass onto CodeHinter which is subscribed to state directly. We can remove currentState from here.

File: frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/ToolJetDbOperations.jsx [hunk 6]
```diff
@@ -9,12 +9,13 @@ import { toast } from 'react-hot-toast';
...
 import { useMounted } from '@/_hooks/use-mount';
+import { useCurrentState } from '@/_stores/currentStateStore';
 
-const ToolJetDbOperations = ({ currentState, optionchanged, options, darkMode }) => {
+const ToolJetDbOperations = ({ optionchanged, options, darkMode }) => {
   const computeSelectStyles = (darkMode, width) => {
     return queryManagerSelectComponentStyle(darkMode, width);
   };
-
+  const currentState = useCurrentState();
```
Comment (johnsoncherian) [inline frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/ToolJetDbOperations.jsx:18]: `currentState`` is used here only to pass down to child components ListRows, CreateRow, UpdateRows and DeleteRows where we have already removed the dependancy. So we can remove currentState from this component as well.

**Pull request merged**